Say-Their-Name/say-their-names-web
# Setup Google Analytics Page Tracking

#### What does this PR do?
- Adds ```react-ga``` package to handle google analytics
- Creates utility functions using said package for tracking:
  - initializeGA: start google analytics and load script (happens before components load)
  - trackInitialPageView: tracks the initial load (using ```useState`` for this and a flag, since the regular implementation doesn't handle the first page load)
  - trackPageView: tracks every page load after the initial one (using the ```useEffect``` hook and firing every time ```history``` changes. 
  - trackEvent: tracks custom events (not yet in use, but i made it so if someone else has to add event tracking, the function body is setup)

#### Description of Task to be completed?
- Add google analytics to all pages (tracks page views for each page including each individual person page.) 

#### How should this be manually tested?
-  Navigate to localhost:3000 once the build is running
-  Check the network tab for requests made to ```https://www.google-analytics.com/g/collect```. Will sometimes take ~2 seconds to show
- Verify the query string parameters in said request are correct: [Parameter Docs Here](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#ea):
- Repeat for every route in routes/index.js
- Check google analytics panel if you have access to verify the requests have been saved properly
- You can additionally run ```yarn test``` to see which events are being fired in the console
 
#### What are the relevant Trello Card stories? [ Trello Card Item ](https://trello.com/b/8x3mo2V5/say-their-names-web)
- https://trello.com/c/smTkEZiO/91-add-google-analytics-to-all-pages

#### Pull Request Type (check one)

#### Developer Checklist

#### Screenshots (if appropriate)
![Screen Shot 2020-06-06 at 6 33 20 PM](https://user-images.githubusercontent.com/6787010/83955969-a47b7780-a826-11ea-8bd2-653579e7adb3.png)
![Screen Shot 2020-06-06 at 6 33 40 PM](https://user-images.githubusercontent.com/6787010/83955971-a6ddd180-a826-11ea-9600-3c471865ea94.png)
![Screen Shot 2020-06-06 at 6 33 56 PM](https://user-images.githubusercontent.com/6787010/83955973-a9402b80-a826-11ea-8793-d94db6bf80eb.png)
